### PR TITLE
Docs shorthand object notation

### PR DESCRIPTION
**Pull request by danielbayley**
A simple but slightly more imaginative example of shorthand object notation.

Closes #1808.

### CODE REVIEW

File: documentation/index.html.erb [hunk 1]
```diff
@@ -418,6 +418,11 @@ Expressions
       about it (say, when using jQuery).
     </p>
     <%= code_for('objects_reserved') %>
+    <p>
+      CoffeeScript has a shortcut for creating objects when you want the key
+      to be set with a variable of the same name.
+    </p>
+    <%= code_for('objects_shorthand') %>
```
Comment (lydell) [inline documentation/index.html.erb]: This file (documentation/index.html.erb) was renamed almost three years ago.

Comment (danielbayley) [inline documentation/index.html.erb]: Ah, my local commit didn't even have this in the commit, as I had just done `git fetch origin pull/1808/head:docs-shorthand-object-notation`

**Comment by lydell** [comment]
@danielbayley I think it’s easiest to start over from the latest master :)

**Comment by danielbayley** [comment]
@lydell You're right 😄 So much for trying to do it the "proper" way…

File: documentation/coffee/objects_shorthand.coffee [hunk 2]
```diff
@@ -0,0 +1,6 @@
+pizzas = 4
```
Comment (lydell) [inline documentation/coffee/objects_shorthand.coffee]: No need for pizzas.

File: documentation/js/objects_shorthand.js [hunk 3]
```diff
@@ -0,0 +1,14 @@
...
+mask = "Orange";
+weapon = "Nunchuks";
+
+turtle = {
+  name: name,
+  mask: mask,
+  weapon: weapon
+};
+
+output = turtle.name + " wears an " + turtle.mask + " mask.";
```
Comment (lydell) [inline documentation/js/objects_shorthand.js]: Please compile this file from objects_shorthand.coffee. Currently, it is out-of-sync.

File: documentation/coffee/objects_shorthand.coffee (hunk 2 above)
Comment (GeoffreyBooth) [inline documentation/coffee/objects_shorthand.coffee]: Best comment ever.

File: documentation/coffee/objects_shorthand.coffee [hunk 4]
```diff
@@ -0,0 +1,6 @@
...
+name = "Michelangelo"
+mask = "Orange"
+weapon = "Nunchuks"
+turtle = {name, mask, weapon}
+output = "#{turtle.name} wears an #{turtle.mask} mask."
```
Comment (lydell) [inline documentation/coffee/objects_shorthand.coffee]:
```coffee
name = "Michelangelo"
mask = "orange"
weapon = "nunchuks"
turtle = {name, mask, weapon}
output = "#{turtle.name} wears an #{turtle.mask} mask. Look out for his #{turtle.weapon}!"
```

better?

**Comment by GeoffreyBooth** [comment]
@danielbayley You’re right, `cake doc:site` should be watching the example source files. I made that update in `master`, along with some general reorganization, and merged those changes into your branch. I also made the update that @lydell was requesting, so I think this PR should now be good to go?

**Pull request merged**